File: app/src/screens/AvatarPickerScreen.js
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet, ScrollView } from 'react-native';
import { colors, spacing, radius } from '../theme';
import { CATEGORIES, presetsByCategory } from '../avatars';
import { api } from '../api';
import { useAuth, refreshUser } from '../auth';

// Tek hazır avatar kutusu
function AvatarOptionCard({ preset, selected, onPress }) {
  return (
    <TouchableOpacity style={styles.cardWrap} onPress={onPress} activeOpacity={0.8}>
      <View style={[styles.circle, { backgroundColor: preset.bg }, selected && styles.circleOn]}>
        <Text style={styles.emoji}>{preset.emoji}</Text>
      </View>
      <Text style={styles.cardLabel} numberOfLines={1}>{preset.label}</Text>
    </TouchableOpacity>
  );
}

// Bir kategorinin avatar ızgarası
function AvatarGrid({ category, selectedKey, onSelect }) {
  const items = presetsByCategory(category.key);
  if (!items.length) return null;
  return (
    <View style={styles.section}>
      <Text style={styles.catTitle}>{category.label}</Text>
      <View style={styles.grid}>
        {items.map((p) => (
          <AvatarOptionCard key={p.key} preset={p} selected={selectedKey === p.key} onPress={() => onSelect(p.key)} />
        ))}
      </View>
    </View>
  );
}

export default function AvatarPickerScreen({ navigation }) {
  const { user } = useAuth();
  const onSelect = async (key) => {
    try { await api.updateProfile({ avatarType: 'preset', avatarKey: key }); await refreshUser(); } catch (e) { alert(e.message); }
    navigation.goBack();
  };
  return (
    <ScrollView style={styles.container} contentContainerStyle={{ padding: spacing.lg, paddingBottom: spacing.xl }}>
      <Text style={styles.note}>Futbol temalı hazır avatarlar. Birine dokun, profilin güncellensin.</Text>
      {CATEGORIES.map((c) => (
        <AvatarGrid key={c.key} category={c} selectedKey={user?.avatar_type === 'preset' ? user.avatar_key : null} onSelect={onSelect} />
      ))}
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: colors.bg },
  note: { color: colors.textMuted, fontSize: 13, marginBottom: spacing.md },
  section: { marginBottom: spacing.lg },
  catTitle: { color: colors.text, fontSize: 15, fontWeight: '800', marginBottom: spacing.sm },
  grid: { flexDirection: 'row', flexWrap: 'wrap', gap: spacing.md },
  cardWrap: { width: 76, alignItems: 'center' },
  circle: { width: 58, height: 58, borderRadius: 29, alignItems: 'center', justifyContent: 'center', borderWidth: 2, borderColor: 'transparent' },
  circleOn: { borderColor: colors.primary },
  emoji: { fontSize: 30, lineHeight: 38 },
  cardLabel: { color: colors.textMuted, fontSize: 10.5, fontWeight: '600', marginTop: 5, textAlign: 'center' },
});

File: lib/features/profile/avatar_picker_screen.dart
// KAYNAK: app/src/screens/AvatarPickerScreen.js — BİREBİR çeviri.
//
// Hazır avatarlar kategori kategori ızgarada listelenir; birine dokununca
// profil sunucuda güncellenir, kullanıcı tazelenir ve ekran kapanır.

import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';

import '../../core/auth.dart';
import '../../core/network/api_client.dart';
import '../../core/theme/tokens.dart';
import '../../widgets/avatar.dart';

class AvatarPickerScreen extends StatelessWidget {
  const AvatarPickerScreen({super.key});

  Future<void> _sec(BuildContext context, String key) async {
    try {
      await api.updateProfile({'avatarType': 'preset', 'avatarKey': key});
      await refreshUser();
    } catch (e) {
      // Kaynakta `alert(e.message)`. Flutter'da karşılığı SnackBar.
      // `ApiException.toString()` zaten yalın mesajı döndürür — kullanıcıya
      // "Exception: ..." ön eki GÖSTERİLMEZ.
      if (context.mounted) {
        ScaffoldMessenger.of(
          context,
        ).showSnackBar(SnackBar(content: Text('$e')));
      }
    }
    // Kaynakta goBack() hata durumunda da çağrılıyor — aynen korunur.
    if (context.mounted) context.pop();
  }

  @override
  Widget build(BuildContext context) => Scaffold(
    backgroundColor: AppColors.bg,
    appBar: AppBar(title: const Text('Avatar Seç')),
    body: ValueListenableBuilder<AuthState>(
      valueListenable: authState,
      builder: (context, s, _) {
        // Yalnız 'preset' tipinde seçili anahtar vardır; yüklenmiş fotoğraf
        // kullanan kullanıcıda hiçbir kutu seçili görünmez (kaynak aynen).
        // Parantez ŞART: `as String?` içindeki `?` parantezsiz yazılınca
        // Dart onu üçlü işlecin `:` dalı sanır ve dosya derlenmez.
        final secili = s.user?['avatar_type'] == 'preset'
            ? (s.user?['avatar_key'] as String?)
            : null;
        return ListView(
          padding: const EdgeInsets.fromLTRB(
            Spacing.lg,
            Spacing.lg,
            Spacing.lg,
            Spacing.xl,
          ),
          children: [
            const Text(
              'Futbol temalı hazır avatarlar. Birine dokun, profilin güncellensin.',
              style: TextStyle(color: AppColors.textMuted, fontSize: 13),
            ),
            const SizedBox(height: Spacing.md),
            for (final c in kAvatarCategories)
              _Izgara(
                kategori: c,
                seciliKey: secili,
                onSelect: (k) => _sec(context, k),
              ),
          ],
        );
      },
    ),
  );
}

class _Izgara extends StatelessWidget {
  const _Izgara({
    required this.kategori,
    required this.seciliKey,
    required this.onSelect,
  });

  final AvatarKategori kategori;
  final String? seciliKey;
  final void Function(String) onSelect;

  @override
  Widget build(BuildContext context) {
    final items = presetsByCategory(kategori.key);
    if (items.isEmpty) return const SizedBox.shrink();
    return Padding(
      padding: const EdgeInsets.only(bottom: Spacing.lg),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            kategori.label,
            style: const TextStyle(
              color: AppColors.text,
              fontSize: 15,
              fontWeight: AppFont.bold,
            ),
          ),
          const SizedBox(height: Spacing.sm),
          Wrap(
            spacing: Spacing.md,
            runSpacing: Spacing.md,
            children: [
              for (final p in items)
                _Kutu(
                  preset: p,
                  selected: seciliKey == p.key,
                  onTap: () => onSelect(p.key),
                ),
            ],
          ),
        ],
      ),
    );
  }
}

class _Kutu extends StatelessWidget {
  const _Kutu({
    required this.preset,
    required this.selected,
    required this.onTap,
  });

  final AvatarPreset preset;
  final bool selected;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) => GestureDetector(
    onTap: onTap,
    child: SizedBox(
      width: 76,
      child: Column(
        children: [
          Container(
            width: 58,
            height: 58,
            alignment: Alignment.center,
            decoration: BoxDecoration(
              color: preset.bg,
              shape: BoxShape.circle,
              border: Border.all(
                width: 2,
                // Seçili değilken kaynakta 'transparent' — kenarlık payı
                // korunur ki seçince kutu ZIPLAMASIN.
                color: selected ? AppColors.primary : Colors.transparent,
              ),
            ),
            child: Text(
              preset.emoji,
              style: const TextStyle(fontSize: 30, height: 1.27),
            ),
          ),
          const SizedBox(height: 5),
          Text(
            preset.label,
            maxLines: 1,
            overflow: TextOverflow.ellipsis,
            textAlign: TextAlign.center,
            style: const TextStyle(
              color: AppColors.textMuted,
              fontSize: 10.5,
              fontWeight: AppFont.semibold,
            ),
          ),
        ],
      ),
    ),
  );
}
